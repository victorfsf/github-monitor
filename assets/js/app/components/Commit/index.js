import Commit from './Commit';

export default Commit;
