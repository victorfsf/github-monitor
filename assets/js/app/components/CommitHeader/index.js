import CommitHeader from './CommitHeader';

export default CommitHeader;
