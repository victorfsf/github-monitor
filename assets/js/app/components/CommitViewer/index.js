import CommitViewer from './CommitViewer';

export default CommitViewer;
