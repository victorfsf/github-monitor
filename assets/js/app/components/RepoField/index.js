import RepoField from './RepoField';

export default RepoField;
