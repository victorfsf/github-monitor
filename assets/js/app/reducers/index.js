import { combineReducers } from 'redux';

// TODO: write actual reducers
const mockedReducer = (state, action) => ({ mock: action });

const rootReducer = combineReducers({
  mockedReducer,
});

export default rootReducer;
