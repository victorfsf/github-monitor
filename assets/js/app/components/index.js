import Box from './Box';
import CommitViewer from './CommitViewer';
import Container from './Container';
import LoginForm from './LoginForm';
import RepoField from './RepoField';
import Spinner from './Spinner';
import Pagination from './Pagination';
import NotFound from './NotFound';

export {
  Box,
  CommitViewer,
  Container,
  LoginForm,
  RepoField,
  Spinner,
  Pagination,
  NotFound,
};
