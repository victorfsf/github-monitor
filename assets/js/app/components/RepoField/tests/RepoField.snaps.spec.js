import React from 'react';
import renderer from 'react-test-renderer';
import { RepoField } from 'app/components';


describe('RepoField', () => {
  let Component;
  let tree;

  test('does not render', () => {
    global.console = {
      error: jest.fn(),
    };
    renderer.create(<RepoField />);
    renderer.create(<RepoField isFetching value="test" />);
    renderer.create(<RepoField hasError errorMessage="error" />);
    renderer.create(<RepoField value="test" />);
    expect(global.console.error).toHaveBeenCalledTimes(1);
  });

  test('renders', () => {
    Component = renderer.create(
      <RepoField onChange={e => e} />,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders with value', () => {
    Component = renderer.create(
      <RepoField value="test/test" onChange={e => e} />,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders with error', () => {
    Component = renderer.create(
      <RepoField
        value="test/test"
        onChange={e => e}
        errorMessage="Repository not found."
        hasError
      />,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders while fetching', () => {
    Component = renderer.create(
      <RepoField
        value="test/test"
        onChange={e => e}
        isFetching
      />,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
