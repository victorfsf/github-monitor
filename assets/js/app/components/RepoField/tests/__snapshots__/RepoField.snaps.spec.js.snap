// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoField renders 1`] = `
<div
  className="form-group"
>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <span
        className="input-group-text"
      >
        <i
          className="fa fa-github-alt"
        />
      </span>
    </div>
    <input
      aria-describedby="addRepo"
      autoComplete="off"
      className="form-control"
      id="id_name"
      name="name"
      onChange={[Function]}
      placeholder="repository-owner/repository-name"
      type="text"
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-info"
        disabled={false}
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Repository URL:
    <a
      className="ml-1"
      href="https://github.com/repository-owner/repository-name"
      target="blank_"
    >
      https://github.com/repository-owner/repository-name
    </a>
  </small>
</div>
`;

exports[`RepoField renders while fetching 1`] = `
<div
  className="form-group"
>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <span
        className="input-group-text"
      >
        <i
          className="fa fa-github-alt"
        />
      </span>
    </div>
    <input
      aria-describedby="addRepo"
      autoComplete="off"
      className="form-control"
      id="id_name"
      name="name"
      onChange={[Function]}
      placeholder="repository-owner/repository-name"
      type="text"
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-info"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Repository URL:
    <a
      className="ml-1"
      href="https://github.com/test/test"
      target="blank_"
    >
      https://github.com/test/test
    </a>
  </small>
</div>
`;

exports[`RepoField renders with error 1`] = `
<div
  className="form-group"
>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <span
        className="input-group-text"
      >
        <i
          className="fa fa-github-alt"
        />
      </span>
    </div>
    <input
      aria-describedby="addRepo"
      autoComplete="off"
      className="form-control is-invalid"
      id="id_name"
      name="name"
      onChange={[Function]}
      placeholder="repository-owner/repository-name"
      type="text"
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-info"
        disabled={false}
        type="submit"
      >
        Submit
      </button>
    </div>
    <div
      className="invalid-feedback"
    >
      Repository not found.
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Repository URL:
    <a
      className="ml-1"
      href="https://github.com/test/test"
      target="blank_"
    >
      https://github.com/test/test
    </a>
  </small>
</div>
`;

exports[`RepoField renders with value 1`] = `
<div
  className="form-group"
>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <span
        className="input-group-text"
      >
        <i
          className="fa fa-github-alt"
        />
      </span>
    </div>
    <input
      aria-describedby="addRepo"
      autoComplete="off"
      className="form-control"
      id="id_name"
      name="name"
      onChange={[Function]}
      placeholder="repository-owner/repository-name"
      type="text"
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-info"
        disabled={false}
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Repository URL:
    <a
      className="ml-1"
      href="https://github.com/test/test"
      target="blank_"
    >
      https://github.com/test/test
    </a>
  </small>
</div>
`;
